### PR TITLE
Add backpack downloading Inital attempt

### PR DESCRIPTION
**Pull request by Arzte**
Functional, but needs polish.

Especially some eye towards how it renders on mobile is needed. Beyond my skill level though.

### CODE REVIEW

**Comment by Arzte** [comment]
I've hacked on consistent sizing, which currently involves manually setting the div width to the same size as the canvas width (`1280px`) when rendering. It's certainly not the cleanest way possible with `modern-screenshot` but it works well.

Still would like to polish this more before merging, but feel free to take over. I'm not going to force-push this branch.

**Comment by Arzte** [comment]
Rebased Pr on master so toasts can be used. They are currently obstructed by modal dialog, but otherwise function properly.

**Comment by Arzte** [comment]
Rebased on master, made a couple QOL changes.

File: components/viewer/modal/BackpackModal.vue [hunk 1]
```diff
@@ -114,6 +135,85 @@ const objectMode = computed(() => {
...
+const isLoading = ref(false);
+const backpackToImage = async () => {
+  if (backpackRef.value && packRows.value.length >= 1) {
+    isLoading.value = true;
+    const $toast = useToast();
+    const toastGenerateImage = $toast.info('Generating image...', {
+      timeout: false,
+    });
+    // A hack, the DOM won't update until after the image is generated otherwise
+    const pause = new Promise((resolve) => setTimeout(resolve, 200));
```
Comment (ltouroumov) [inline components/viewer/modal/BackpackModal.vue:149]: There's a `sleep` function in `utils.ts`

iirc, `await nextTick()` should work here.

**Review by ltouroumov — APPROVED** [review]

File: components/viewer/modal/BackpackModal.vue (hunk 1 above)
Comment (Arzte) [reply components/viewer/modal/BackpackModal.vue:149]: nextTick works, the sleep was leftover from my attempt with modern-screenshot.

**Review by ltouroumov — APPROVED** [review]

**Pull request merged**